### PR TITLE
Add `keys delete` command

### PR DESCRIPTION
**Pull request by hu55a1n1**
Closes: #1065

## Description

Adds the `keys delete` command to the Hermes CLI:

```bash
hermes keys delete <chain-id> [-n|--name <keyname>]
```

### Usage

```
USAGE:
    hermes keys delete <OPTIONS>

DESCRIPTION:
    Delete key(s) from a configured chain

POSITIONAL ARGUMENTS:
    chain_id                  identifier of the chain

FLAGS:
    -n, --name NAME           name of the key
    -a, --all                 delete all keys
```
______

For contributor use:

- [x] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Updated relevant documentation (`docs/`) and code comments.
- [x] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

File: relayer-cli/src/commands/keys/clean.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+#[derive(Clone, Command, Debug, Options)]
+pub struct KeysCleanCmd {
+    #[options(free, required, help = "identifier of the chain")]
+    chain_id: ChainId,
+
+    #[options(
+        short = "n",
+        help = "name of the key (defaults to the `key_name` defined in the config)"
+    )]
+    name: Option<String>,
```
Comment (hu55a1n1) [inline relayer-cli/src/commands/keys/clean.rs:22]: Should we also add an optional `--all` flag that deletes all keys for a chain?

File: relayer-cli/src/commands/keys.rs [hunk 2]
```diff
@@ -16,6 +17,10 @@ pub enum KeysCmd {
     #[options(help = "Adds a key to a configured chain")]
     Add(add::KeysAddCmd),
 
+    /// The `keys clean` subcommand
+    #[options(help = "Removes a key from a configured chain")]
+    Clean(clean::KeysCleanCmd),
```
Comment (hu55a1n1) [inline relayer-cli/src/commands/keys.rs:22]: Should we name the command delete instead of clean? Just to stay consistent with `gaiad`?

Comment (adizere) [reply relayer-cli/src/commands/keys.rs:22]: I agree. Additionally, `delete` would be more accurate than `clean`.

File: relayer-cli/src/commands/keys/clean.rs (hunk 1 above)
Comment (adizere) [reply relayer-cli/src/commands/keys/clean.rs:22]: Good idea.

If I were a user, I would rather not have any default here. I would prefer to _always_ specify which key to delete (or all keys). So I would suggest we can even remove the part "(defaults to the `key_name` defined in the config)". Then this CLI would expect one out of two options


- `--name/-n KEY-NAME` or
- `--all/-a`

Comment (hu55a1n1) [reply relayer-cli/src/commands/keys/clean.rs:22]: Done... 👍

**Review by adizere — APPROVED** [review]
Looks good!

**Comment by adizere** [comment]
This is ready for review @andynog and @greg-szabo.

File: relayer/src/keyring.rs [hunk 3]
```diff
@@ -136,6 +137,11 @@ impl KeyStore for Memory {
         }
     }
 
+    fn remove_key(&mut self, key_name: &str) -> Result<(), Error> {
+        self.keys.remove(key_name).ok_or(Error::key_not_found())?;
```
Comment (romac) [inline relayer/src/keyring.rs:141]:
```suggestion
        self.keys.remove(key_name).ok_or(Error::key_not_found)?;
```

File: relayer/src/keyring.rs [hunk 4]
```diff
@@ -138,7 +138,7 @@ impl KeyStore for Memory {
     }
 
     fn remove_key(&mut self, key_name: &str) -> Result<(), Error> {
-        self.keys.remove(key_name).ok_or(Error::key_not_found())?;
+        self.keys.remove(key_name).ok_or(Error::key_not_found)?;
```
Comment (adizere) [inline relayer/src/keyring.rs:141]: It seems like it was correct before.

Comment (romac) [reply relayer/src/keyring.rs:141]: It wasn't but the fix suggested by the compiler is even worse. Will fix it manually.

Comment (romac) [reply relayer/src/keyring.rs:141]: My bad, I actually misapplied the compiler suggestion.

Comment (romac) [reply relayer/src/keyring.rs:141]: Fixed

Comment (romac) [reply relayer/src/keyring.rs]: Fixed

**Pull request merged**